### ISSUE TITLE
[CORE] Set ray client server host when starting ray.

### ISSUE THREAD

**Issue by surenyufuz** [ray-project/ray#48836]
### Description

When starting ray processes, we can use the flag `--dashboard-host` to set the host to bind for the dashboard server, either localhost (127.0.0.1) or 0.0.0.0.

**Comment by Superskyyy** [comment]
You mean in the case that you will only dispatch from within the head node so there isn't a need to expose the client server host to the public right?

**Comment by surenyufuz** [comment]
> You mean in the case that you will only dispatch from within the head node so there isn't a need to expose the client server host to the public right?

Thanks for your reply.  Maybe I can use an exposed proxy with authentication logic to check whether a user could connect to the ray client server, the proxy and the head node are deployed on  the same Kubernetes node,  both of them could connect to each other by the host net.

**Comment by jjyao** [comment]
@surenyufuz Are you trying to use Ray client from within the Ray cluster?

**Comment by surenyufuz** [comment]
> @surenyufuz Are you trying to use Ray client from within the Ray cluster?

No, I am trying to use Ray client in a Kubernetes pod outside the Ray cluster through a proxy. The exposed proxy is deployed with the head node on a Kubernetes  node, so I need not to expose the client server host to the public.
